packages/app/background-jobs-common/src/background-job-processor/processors/AbstractBackgroundJobProcessor.ts

```diff
@@ -237,11 +237,31 @@ export abstract class AbstractBackgroundJobProcessor<
   }
 
   public async schedule(jobData: JobPayload, options?: JobOptionsType): Promise<string> {
-    const jobIds = await this.scheduleBulk([jobData], options)
-    return jobIds[0]
+    await this.startIfNotStarted()
+
+    const job = await this._queue?.add(
+      this.config.queueId,
+      jobData,
+      prepareJobOptions(this.config.isTest, options),
+    )
+    if (!job?.id) {
+      throw new Error('Scheduled job ID is undefined')
+    }
+
+    if (this._spy) {
+      this._spy.addJob(job, 'scheduled')
+    }
+
+    return job.id
   }
 
   public async scheduleBulk(jobData: JobPayload[], options?: JobOptionsType): Promise<string[]> {
+    if (options?.repeat) {
+      throw new Error(
+        'scheduleBulk does not support repeatOptions. Please use schedule method instead',
+      )
+    }
+
```

Comment:
If `repeat` does not work with the bulk method, instead of throwing an error should we just exclude it from the type?

Reply:
wouldn't TS still accept jobs with unnecessary fields?

Reply:
+1 on what Igor said. That's basically what was happening but further when we were calling BullMQ, it was accepting `repeat` settings but not using them since they're excluded, and it's misleading because you don't get any error but your job is scheduled as a regular one, not a repeatable one 😕

Reply:
TS will accept it, you are right, but if it is not part of API as a user I would expect it to not affect the method which is what is currently happening (bullMQ is ignoring it)

Reply:
Actually, won't TS complain if we pass a not-allowed prop 🤔 
<img width="1195" alt="Screenshot 2024-08-21 at 15 19 08" src="https://github.com/user-attachments/assets/5139c877-062c-4351-b32e-bd0a541d4471">

In any case, it is still possible to pass it but as a developer, if it is not defined on API I won't expect it to work.

That being said if you think that the error is a better approach, I have no strong opinion, and feel free to go with this approach if it looks better for you 🙇

Reply:
I think you're right @CarlosGamero. I didn't notice the TS error yesterday, but it's definitely there, so that should be enough to discourage wrong usage imo. Let me update this

Reply:
awesome, even better!